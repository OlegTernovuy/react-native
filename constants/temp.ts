const OCCUPATION_TYPE = ['Development', 'Testing', 'Engineering'];

const mockData = [
    {
        name: 'Raquel Powlowski',
        avatar: 'https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/672.jpg',
        occupation: 'Development',
        responsibilitiesAtWork:
            'Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals',
        id: '1',
    },
];

export { OCCUPATION_TYPE, mockData };
