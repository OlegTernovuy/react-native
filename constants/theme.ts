export const THEME = {
    Colors: {
        primary: '#312651',
        secondary: '#444262',

        lightGray: '#f1f1f1cc',
        lightBlue: '#b1d0ffcc',

        lightWhite: '#ffffff',
    },
    Sizes: {
        small: '4px',
        medium: '8px',
        large: '16px',
        xLarge: '20px',
        xxLarge: '24px',
    },
};
