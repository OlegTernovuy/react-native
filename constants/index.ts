import { OCCUPATION_TYPE } from './temp';
import { THEME } from './theme';

export { OCCUPATION_TYPE, THEME };
