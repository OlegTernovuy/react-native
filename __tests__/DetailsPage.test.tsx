import { fireEvent, render } from '@testing-library/react-native';
import { waitFor } from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';
import { Linking } from 'react-native';
import { router } from 'expo-router';
import axios from 'axios';

import CardDetails from '../app/details/[id]';

import { mockData } from '../constants/temp';

const mock = new MockAdapter(axios);

jest.mock('expo-router', () => ({
    useGlobalSearchParams: jest.fn().mockReturnValue({ id: '1' }),
    router: {
        back: jest.fn(),
    },
    Stack: {
        Screen: ({ children, options }: any) => (
            <>
                {options?.headerLeft && options.headerLeft()}
                {children}
            </>
        ),
    },
}));

describe('CardDetaild component', () => {
    afterEach(() => {
        mock.reset();
    });

    it('should matches snapshot when data is loaded', async () => {
        mock.onGet().reply(200, mockData);
        const { toJSON } = render(<CardDetails />);
        await waitFor(() => {
            expect(toJSON()).toMatchSnapshot();
        });
    });

    it('should show loader indicator when data is fetching', () => {
        mock.onGet('', { params: { id: '1' } }).reply(200, mockData);
        const { getByTestId } = render(<CardDetails />);
        expect(getByTestId('activity-indicator')).toBeTruthy();
    });

    it('should render card details when data is fetched', async () => {
        mock.onGet('', { params: { id: '1' } }).reply(200, mockData);
        const { getByText } = render(<CardDetails />);
        await waitFor(() => {
            expect(getByText(mockData[0].name)).toBeTruthy();
        });
    });

    it('should call router.back on button click', () => {
        mock.onGet('', { params: { id: '1' } }).reply(200, mockData);
        const { getByTestId } = render(<CardDetails />);
        fireEvent.press(getByTestId('back-button'));
        expect(router.back).toHaveBeenCalledTimes(1);
    });

    it('should call open LinkedIn url on button click save employees', () => {
        mock.onGet('', { params: { id: '1' } }).reply(200, mockData);
        const openURL = jest.spyOn(Linking, 'openURL');
        const { getByTestId } = render(<CardDetails />);
        fireEvent.press(getByTestId('save-emp-button'));
        expect(openURL).toHaveBeenCalledWith('https://www.linkedin.com/feed/');
    });
});
