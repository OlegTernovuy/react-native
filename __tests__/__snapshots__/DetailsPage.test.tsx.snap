// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetaild component should matches snapshot when data is loaded 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#ffffff",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingBottom": 0,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="back-button"
  >
    <Text />
  </View>
  <View>
    <ActivityIndicator
      testID="activity-indicator"
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "bottom": 8,
        "height": 75,
        "left": 0,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#b1d0ffcc",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="save-emp-button"
    >
      <Text
        style={
          {
            "color": "#312651",
            "fontSize": 20,
            "fontWeight": "600",
          }
        }
      >
        Save Employees
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
